packages/xstate-react/test/types.test.tsx

```diff
@@ -114,4 +110,18 @@ describe('useMachine', () => {
 
     render(<YesNo />);
   });
+
+  it('state should not become never after checking state with matches', () => {
+    const YesNo = () => {
+      const [state] = useMachine<any, any>(yesNoMachine);
+
+      if (state.matches('no')) {
+        return <span>No</span>;
```

Comment:
The problem seems to happen right when returning right after using `state.matches`. 

Replacing this line with `console.log(state.context.value)` works fine.

In fact, even the following code works fine. 😕 Maybe another TypeScript regression? 

```JavaScript
const isNo = state.matches('no');
if (isNo) {
  return <span>No</span>;
}
```